File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/data/repository/GymRepository.kt
package com.gym.gymapp.data.repository

import com.gym.gymapp.data.models.*
import com.gym.gymapp.network.ApiEndpoints
import com.gym.gymapp.network.NetworkClient
import io.ktor.client.call.*
import io.ktor.client.request.*
import io.ktor.http.*

class GymRepository {
    private val client = NetworkClient.client
    
    suspend fun getGyms(): Result<List<Gym>> {
        return try {
            val response: ApiResponse<List<Gym>> = client.get(ApiEndpoints.GYMS).body()
            Result.success(response.data)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun getDashboardStats(gymId: String): Result<DashboardStats> {
        return try {
            val response: ApiResponse<DashboardStats> = client.get(ApiEndpoints.DASHBOARD_STATS) {
                parameter("gymId", gymId)
            }.body()
            Result.success(response.data)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun getTodayAttendance(gymId: String): Result<List<Attendance>> {
        return try {
            val response: ApiResponse<List<Attendance>> = client.get(ApiEndpoints.ATTENDANCE_TODAY) {
                parameter("gymId", gymId)
            }.body()
            Result.success(response.data)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun checkIn(request: AttendanceCheckInRequest): Result<Attendance> {
        return try {
            val response: ApiResponse<Attendance> = client.post(ApiEndpoints.ATTENDANCE) {
                contentType(ContentType.Application.Json)
                setBody(request)
            }.body()
            Result.success(response.data)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun getMembershipPlans(gymId: String): Result<List<MembershipPlan>> {
        return try {
            val response: ApiResponse<List<MembershipPlan>> = client.get(ApiEndpoints.PLANS) {
                parameter("gymId", gymId)
            }.body()
            Result.success(response.data)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}

File: gym_flutter_app/lib/data/repositories/gym_repository.dart
import 'package:dio/dio.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'api_client.dart';
import '../models/other_models.dart';

final gymRepositoryProvider = Provider<GymRepository>((ref) {
  return GymRepository(ref.watch(apiClientProvider));
});

class GymRepository {
  final Dio _client;

  GymRepository(this._client);

  Future<ApiResponse<List<Gym>>> getGyms() async {
    try {
      final response = await _client.get('/gyms');
      return ApiResponse.fromJson(
        response.data,
        (json) => (json as List).map((e) => Gym.fromJson(e)).toList(),
      );
    } catch (e) {
      return ApiResponse(status: false, message: e.toString());
    }
  }

  Future<ApiResponse<DashboardStats>> getDashboardStats({required String gymId}) async {
    try {
      final response = await _client.get('/dashboard/stats', queryParameters: {'gymId': gymId});
      return ApiResponse.fromJson(
        response.data,
        (json) => DashboardStats.fromJson(json),
      );
    } catch (e) {
      return ApiResponse(status: false, message: e.toString());
    }
  }

  Future<ApiResponse<List<Plan>>> getMembershipPlans({required String gymId}) async {
    try {
      final response = await _client.get('/plans', queryParameters: {'gymId': gymId});
      return ApiResponse.fromJson(
        response.data,
        (json) => (json as List).map((e) => Plan.fromJson(e)).toList(),
      );
    } catch (e) {
      return ApiResponse(status: false, message: e.toString());
    }
  }
}
